jets/e/sha1.c

```diff
@@ -0,0 +1,62 @@
+/* j/5/sha1.c
+**
+*/
+#include "all.h"
+
+
+#if defined(U3_OS_osx)
+#include <CommonCrypto/CommonDigest.h>
+#else
+#include <openssl/sha.h>
+#endif
+
+/* functions
+*/
+
+  u3_noun
+  u3qe_sha1(u3_atom wid,
+            u3_atom dat)
+  {
+    c3_assert(_(u3a_is_cat(wid)));
+    dat = u3qc_rev(3, wid, dat);
+
+    //NOTE XX fixme we shouldn't c3_malloc in inner road,
+    //     this is also done in other SHA jets
+    c3_y* fat_y = c3_malloc(wid + 1);
```

Comment:
Please replace this with `u3a_malloc()`. We might as well fix the other SHA jets here too.

Reply:
Did you also switch to `u3a_free()`?